### ISSUE TITLE
World Conquest leaks preprocessor definitions such as `STR_CASTLE` in the editor

### ISSUE THREAD

**Issue by stevecotton** [wesnoth/wesnoth#6062]
### Game and System Information
Seen on 1.15.17+dev, shortly after the 1.15.17 tag (local Linux build from 5bdae673af36856a8b65e1ea724bd629564abf3b)

### Describe the bug
World Conquest `#define`s some additional macros with generic names, at least when the editor is started. These clash with similarly overly-generically named defines in add-ons.

### To Reproduce
1. Have the War of Legends era installed (others will be affected, but that's the one which this was noticed with).
2. On the title screen, click on "Map Editor".
3. See the error message:

```
warning preprocessor: Redefining macro STR_CASTLE without explicit #undef at ~add-ons/War_of_Legends/macros/mainline-strings.cfg:17
    included from ~add-ons/War_of_Legends/_main.cfg:63
previously defined at campaigns/World_Conquest/./resources/./strings/mainline.cfg:19
    included from campaigns/World_Conquest/./resources/resources.cfg:4
    included from campaigns/World_Conquest/_main.cfg:16
    included from _main.cfg:38
```

**Comment by knyghtmare** [comment]
This probably triggered since WoL is one of the eras to allow placement of its units within the map editor.

**Comment by stevecotton** [comment]
Looks like it will trigger whenever `MULTIPLAYER` is defined too.

As a fix, I'm trying a big search-and-replace to change `#define STR_SOMETHING` to `#define WC_STR_SOMETHING`.

**Comment by Pentarctagon** [comment]
I'll add that I'm still very much not in favor of WC defining these macros in the first place.  This isn't directly related to this issue of course, but WC's code seems to have a tendency to define small/tiny single use macros rather than just putting the WML/text into the file.  For example [WC_]STR_HEROIC_FEMALE - it's used in exactly one place and in this case both files are even in the same textdomain.  There's no reason this macro needs to exist at all.

**Comment by CelticMinstrel** [comment]
I agree, in cases where the string is used only once, it should probably be inlined. But that's not a priority… I can get to it myself, eventually.

**Comment by stevecotton** [comment]
Fully agreed, but I think merging the current PRs is better than not merging them.

**Comment by stevecotton** [comment]
Fixed in #6063 / #6064.
